### PR TITLE
Add npm integration tests

### PR DESCRIPTION
**Pull request by danielbrzn**
This PR aims to add integration tests for Strongbox with npm.

It tests for the following scenarios:

- Publishing a node package to Strongbox
- Resolving a concrete node package via Strongbox

Resolves https://github.com/strongbox/strongbox/issues/433

### CODE REVIEW

File: npm/src/it/common-flow/test-npm-common-flow.groovy [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    loadScriptByName('NpmIntegrationTest.groovy')
+}
+this.metaClass.mixin baseScript
+
+println "Test test-npm-common-flow.groovy" + "\n\n"
+
+def npmExec
+def publishSuccessMsg = "+ @strongbox/hello-strongbox-npm@1.0.0"
+def resolveSuccessMsgWindows = "`-- @strongbox/hello-strongbox-npm@1.0.0"
+def resolveSuccessMsgUnix = "└─┬ @strongbox/hello-strongbox-npm@1.0.0"
```
Comment (SoulSilence) [inline npm/src/it/common-flow/test-npm-common-flow.groovy]: `└─┬` This symbol differs for me (mint 18.3). I suppose we can remove it and use only `@strongbox/hello-strongbox-npm@1.0.0` text for assertion for both systems

Comment (danielbrzn) [reply npm/src/it/common-flow/test-npm-common-flow.groovy]: I'll change this to something that is OS independent: the error message as shown below.

`npm ERR! missing: @strongbox/hello-strongbox-npm`

**Comment by sbespalov** [comment]
@carlspring I think we can merge it
@danielbrzn many thanks for your contribution

**Comment by carlspring** [comment]
Thanks, @danielbrzn and @sbespalov!

**Pull request merged**